repository/src/main/java/org/apache/atlas/repository/graph/TypeCacheRefresher.java

```diff
@@ -53,88 +53,79 @@ public void init() throws AtlasException {
-            refreshCache(cacheRefresherEndpoint,totalFieldKeys);
-        }
-        catch (final Exception exception) {
-            LOG.error(exception.getMessage(),exception);
-            throw new AtlasBaseException(AtlasErrorCode.INTERNAL_ERROR, "Could not update type definition cache");
-        }
+        int totalFieldKeys = provider.get().getManagementSystem().getGraphIndex(VERTEX_INDEX).getFieldKeys().size();
+        LOG.info("Found {} totalFieldKeys to be expected in other nodes :: traceId {}", totalFieldKeys, traceId);
+        refreshCache(totalFieldKeys, traceId);
     }
 
-    private void refreshCache(final String hostUrl,final int totalFieldKeys) {
-        final CloseableHttpClient client = HttpClients.createDefault();
-        CloseableHttpResponse response = null;
+    private void refreshCache(final int totalFieldKeys, final String traceId) throws IOException, URISyntaxException {
+        URIBuilder builder = new URIBuilder(cacheRefresherEndpoint);
+        builder.setParameter("expectedFieldKeys", String.valueOf(totalFieldKeys));
+        builder.setParameter("traceId", traceId);
+        final HttpPost httpPost = new HttpPost(builder.build());
+        LOG.info("Invoking cache refresh endpoint {} :: traceId {}", cacheRefresherEndpoint, traceId);
 
-        try {
-            URIBuilder builder = new URIBuilder(hostUrl);
-            builder.setParameter("expectedFieldKeys", String.valueOf(totalFieldKeys));
-            final HttpPost httpPost = new HttpPost(builder.build());
-            LOG.info("Invoking cache refresh endpoint {}",hostUrl);
-            response = client.execute(httpPost);
-            LOG.info("Received HTTP response code {} from cache refresh endpoint",response.getStatusLine().getStatusCode());
-            if(response.getStatusLine().getStatusCode() != 200) {
-                throw new RuntimeException("Error while calling cache-refresher on host "+hostUrl+". HTTP code = "+ response.getStatusLine().getStatusCode());
-            }
-            final String responseBody = EntityUtils.toString(response.getEntity());
-            LOG.info("Response Body from cache-refresh = {}", responseBody);
-            CacheRefreshResponseEnvelope cacheRefreshResponseEnvelope = convertStringToObject(responseBody);
-
-            for (CacheRefreshResponse responseOfEachNode : cacheRefreshResponseEnvelope.getResponse()) {
-                if (responseOfEachNode.getStatus() != 204) {
-                    //Do not throw exception in this case as node must have been in passive state now
-                    LOG.error("Error while performing cache refresh on host {} . HTTP code = {}", responseOfEachNode.getHost(), responseOfEachNode.getStatus());
-                } else {
-                    LOG.info("Host {} returns response code {}", responseOfEachNode.getHost(), responseOfEachNode.getStatus());
-                }
+        String responseBody;
+        try (CloseableHttpClient client = HttpClients.createDefault()) {
```

Comment:
Previously you were closing the client and response, is that not required now?
IOUtils.closeQuietly(response);
IOUtils.closeQuietly(client);

Reply:
I have used try with resource of java 7  so it will be closed using that feature.